### ISSUE TITLE
Detecting the object used for a method call

### ISSUE THREAD

**Issue by ross-weir** [denoland/deno_lint#523]
In order to improve rules like `no-eval` we need to be able to detect the object a method call is using.

For example when detecting `eval` usages we need to know if the `eval` method is being called specifically using the `window`, `globalThis` or `this` object at a global scope. [Some related discussion on a recent `no-eval` PR.](https://github.com/denoland/deno_lint/pull/490#discussion_r517210917)

We would like to be able to differentiate between:

```js
// Raise a linting warning
this.eval();
globalThis.eval();
window.eval();
```

vs

```js
// Don't raise a linting warning
const o = {
  eval() {}
};
o.eval();
```

I'm not sure how to detect this properly at the moment, suggestions would be greatly appreciated.

Ping @kdy1, I was told you might be able to provide some assistance.

**Comment by kdy1** [comment]
I think it can be implemented using
